src/rules/table-column-count.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @fileoverview Rule to ensure GitHub Flavored Markdown tables have a consistent number of cells in each row.
+ * @author Sweta Tanwar (@SwetaTanwar)
+ */
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @typedef {import("../types.ts").MarkdownRuleDefinition<{ RuleOptions: []; }>} TableColumnCountRuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+/** @type {TableColumnCountRuleDefinition} */
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			recommended: true,
+			description:
+				"Disallow data rows in a GitHub Flavored Markdown table from having more cells than the header row",
+			url: "https://github.com/eslint/markdown/blob/main/docs/rules/table-column-count.md",
+		},
+
+		fixable: null,
+		schema: [],
+
+		messages: {
+			inconsistentColumnCount:
+				"Data row {{dataRowIndex}} (1-indexed) has {{actualCells}} cells, but header has {{expectedCells}} cells (should not exceed header count).",
```

Comment:
I think we can improve the message:

```suggestion
				"Table column count mismatch (Expected: {{expectedCells}}, Actual: {{actualCells}}), extra data will be ignored.",
```